plugins/modules/vmware_content_deploy_ovf_template.py

```diff
@@ -137,65 +154,129 @@ class VmwareContentDeployOvfTemplate(VmwareRestClient):
     def __init__(self, module):
         """Constructor."""
         super(VmwareContentDeployOvfTemplate, self).__init__(module)
-        self.ovf_template_name = self.params.get('ovf_template')
-        self.content_library_name = self.params.get('content_library')
+
+        # Initialize member variables
+        self.module = module
+        self._pyv = PyVmomi(module=module)
+        self._template_service = self.api_client.vcenter.vm_template.LibraryItems
+        self._datacenter_obj = None
+        self._datastore_id = None
+        self._library_item_id = None
+        self._folder_id = None
+        self._host_id = None
+        self._cluster_id = None
+        self._resourcepool_id = None
+        self.result = {}
+
+        # Turn on debug if not specified, but ANSIBLE_DEBUG is set
+        if self.module._debug:
+            self.warn('Enable debug output because ANSIBLE_DEBUG was set.')
+            self.params['log_level'] = 'debug'
+        self.log_level = self.params['log_level']
+        if self.log_level == 'debug':
+            # Turn on debugging
+            self.result['debug'] = {}
+
+        # Get parameters
+        self.template = self.params.get('template')
+        self.library = self.params.get('library')
         self.vm_name = self.params.get('name')
         self.datacenter = self.params.get('datacenter')
         self.datastore = self.params.get('datastore')
+        self.datastore_cluster = self.params.get('datastore_cluster')
         self.folder = self.params.get('folder')
         self.resourcepool = self.params.get('resource_pool')
         self.cluster = self.params.get('cluster')
         self.host = self.params.get('host')
-        self.storage_provisioning = self.params.get('storage_provisioning')
+        self.storage_provisioning = self.params['storage_provisioning']
         if self.storage_provisioning == 'eagerzeroedthick':
             self.storage_provisioning = 'eagerZeroedThick'
 
+        vm = self._pyv.get_vm()
+        if vm:
+            self.result['vm_deploy_info'] = dict(
+                msg="Virtual Machine '%s' already Exists." % self.vm_name,
+                vm_id=vm._moId,
+            )
+            self._fail(msg="Virtual Machine deployment failed")
+
     def deploy_vm_from_ovf_template(self):
         # Find the datacenter by the given datacenter name
-        self.datacenter_id = self.get_datacenter_by_name(datacenter_name=self.datacenter)
-        if not self.datacenter_id:
-            self.module.fail_json(msg="Failed to find the datacenter %s" % self.datacenter)
+        self._datacenter_obj = find_datacenter_by_name(self._pyv.content, datacenter_name=self.datacenter)
```

Comment:
Please use `self.get_datacenter_by_name` instead of `find_datacenter_by_name`

Reply:
Ah good catch! This was needed when I was doing the FQPN processing in-module, before I moved it into vmware.py.